app/commands/payments/paypal/request_access_token.rb

```diff
@@ -0,0 +1,37 @@
+class Payments::Paypal::RequestAccessToken
+  include Mandate
+
+  CACHE_KEY = :paypal_access_token
+
+  def call
+    access_token = Rails.cache.read(CACHE_KEY)
+    return access_token if access_token.present?
+
+    token = request_token!
+    token[:access_token].tap do |new_access_token|
+      expires_in = token[:expires_in].to_i - EXPIRY_TIME_OFFSET
+      Rails.cache.write(CACHE_KEY, new_access_token, expires_in:)
```

Comment:
I'm not sure our production cache supports `expires_in`. Is it needed?

Reply:
I hadn't consider that our Redis cache provider might not support this. 
And it doesn't: https://edgeapi.rubyonrails.org/classes/ActiveSupport/Cache/RedisCacheStore.html

It is not technically needed, we'll just need to request an access token dynamically. It doesn't really influence the flow much at the moment, but we'll be making two request access token calls in the process of creating a paypal subscription instead of one.

Reply:
I really meant, do we need the expires in. Can we just leave it in the cache and leave it to auto-expire.

Reply:
When would it auto-expire? We detect the error response?